core/src/test/java/org/mskcc/oncotree/utils/CacheUtilTest.java

```diff
@@ -128,4 +133,105 @@ public void testRecacheWithBrokenOncoTreeVersionRepository() {
+        for(TumorType tumorType: tumorTypes.values()) {
+            if ("SS".equals(tumorType.getCode())) {
+                assertEquals(1, tumorType.getHistory().size());
+                assertEquals("SEZS", tumorType.getHistory().get(0));
+                foundHistory = Boolean.TRUE;
+                // do not return because we want to check no other history
+            } else {
+                assertEquals(0, tumorType.getHistory().size());
+            }
+            foundHistory = assertHistoryInChildren(tumorType.getChildren(), foundHistory);
+        }
+        return foundHistory;
+    }
+
+    private Boolean assertRescindsInChildren(Map<String, TumorType> tumorTypes, Boolean foundRescinds) {
+        for(TumorType tumorType: tumorTypes.values()) {
+            if ("URMM".equals(tumorType.getCode())) {
+                assertEquals("URMM rescinds size", 1, tumorType.getRescinds().size());
+                assertEquals("GMUCM", tumorType.getRescinds().get(0));
+                foundRescinds = Boolean.TRUE;
+                // do not return because we want to check no other rescinds
+            } else {
+                assertEquals(0, tumorType.getRescinds().size());
+            }
+            foundRescinds = assertRescindsInChildren(tumorType.getChildren(), foundRescinds);
+        }
+        return foundRescinds;
+    }
+
+    private Boolean assertPrecursorsInChildren(Map<String, TumorType> tumorTypes, Boolean foundPrecursors) {
+        for(TumorType tumorType: tumorTypes.values()) {
+            if ("CLLSLL".equals(tumorType.getCode())) {
+                assertEquals("'CLLSLL' precursors size", 2, tumorType.getPrecursors().size());
+                assertTrue("Expected 'CLL' to be precursor to 'CLLSLL'", tumorType.getPrecursors().contains("CLL"));
+                assertTrue("Expected 'SLL' to be precursor to 'CLLSLL'", tumorType.getPrecursors().contains("SLL"));
+                foundPrecursors = Boolean.TRUE;
+                // do not return because we want to check no other precursors
+            } else {
+                assertEquals(0, tumorType.getPrecursors().size());
+            }
+            foundPrecursors = assertPrecursorsInChildren(tumorType.getChildren(), foundPrecursors);
+        }
+        return foundPrecursors;
+    }
+
+    @Test
+    public void testGetTumorTypesByVersion() {
+        Map<String, TumorType> returnedTumorTypes = cacheUtil.getTumorTypesByVersion(legacyVersion);
+        // there is only one node in returnedTumorTypes, and it is the TISSUE node, everything else is a child of that node
+        String rootTumorTypeCode = returnedTumorTypes.keySet().iterator().next();
+        TumorType rootTumorType = returnedTumorTypes.get(rootTumorTypeCode);
+        assertEquals("TISSUE", rootTumorTypeCode);
+        assertEquals("TISSUE", rootTumorType.getCode());
+        assertEquals(32, rootTumorType.getChildren().size());
```

Comment:
32 could be a parameter defined in the config class (or wherever the mock data is) .. or be computed there.
alternatively, we could have a tree traversal to see if the tree returned matches the tree defined in our mock